csrc/runtime/allocations.cpp

```diff
@@ -761,4 +689,64 @@ std::pair<std::vector<int64_t>, std::vector<int64_t>> inferShapeOfOutput(
   return {meta_tensor.sizes().vec(), meta_tensor.strides().vec()};
 }
 
+TensorShapeInfo inferTensorShapes(
+    TensorView* tv,
+    const ExpressionEvaluator& expr_eval) {
+  // Alias handling:
+  auto alias_info = tv->fusion()->getOutputAlias(tv);
+  if (alias_info.type == AllocationType::Evaluate) {
+    auto val = expr_eval.evaluate(tv);
+    NVF_ERROR(val.hasValue() && val.is<at::Tensor>(), "Output is not a Tensor");
```

Comment:
nit：the `hasValue` check is redundant.
```suggestion
    NVF_ERROR(val.is<at::Tensor>(), "Output is not a Tensor");
```

Reply:
I didn't realize that, thanks.

Reply:
Removed redundant check.